Dockerfile

```diff
@@ -34,7 +34,7 @@ ENV CARGO_REGISTRIES_CRATES_IO_PROTOCOL="sparse"
 COPY . .
 
 # Use bash variable substitution to convert environment name to lowercase
-RUN bash -c 'cargo build --release --features ${RUN_ENV@L}'
+RUN bash -c 'cargo build --release --features ${RUN_ENV@L} ${EXTRA_FEATURES@L}'
```

Comment:
You don't need the lowercase transformation here, that was only required in case of `RUN_ENV` due to the way it was handled by the application to load appropriate configuration files.

You'd also need to add an `ARG EXTRA_FEATURES=""` line for the `builder` stage.